mmv1/third_party/terraform/resources/resource_storage_bucket_object.go

```diff
@@ -279,6 +297,43 @@ func resourceStorageBucketObjectCreate(d *schema.ResourceData, meta interface{})
 	return resourceStorageBucketObjectRead(d, meta)
 }
 
+func resourceStorageBucketObjectUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	bucket := d.Get("bucket").(string)
+	name := d.Get("name").(string)
+
+	objectsService := storage.NewObjectsService(config.NewStorageClient(userAgent))
+	getCall := objectsService.Get(bucket, name)
+
+	res, err := getCall.Do()
+
+	if d.HasChange("event_based_hold") {
+		v := d.Get("event_based_hold")
+		res.EventBasedHold = v.(bool)
+	}
+
+	if d.HasChange("temporary_hold") {
+		v := d.Get("temporary_hold")
+		res.TemporaryHold = v.(bool)
+	}
+
+	updateCall := objectsService.Update(bucket, name, res)
+	_, err = updateCall.Do()
+
+	if err != nil {
+		return fmt.Errorf("Error updating object %s: %s", name, err)
+	}
+
+	d.SetId(objectGetID(res))
```

Comment:
object id shouldn't change right? I think this line can be omitted

Reply:
you're right, removed this